### PR TITLE
DM-36673: Add barPlots to analysis_tools

### PR DESCRIPTION
**Pull request by bsmartradio**
Added a bar plot to analysis.tools.plots that can take any number of vector inputs and plot them as one or more bar graphs.

### CODE REVIEW

File: python/lsst/analysis/tools/actions/plot/barPlots.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        for panel in self.panels:  # type: ignore
+            for barData in self.panels[panel].bars.items():  # type: ignore
+                yield barData, Vector
+
+    def __call__(self, data: KeyedData, **kwargs) -> Mapping[str, Figure] | Figure:
+        return self.makePlot(data, **kwargs)
+        # table is a dict that needs: x, y, run, skymap, filter, tract,
```
Comment (sr525) [inline python/lsst/analysis/tools/actions/plot/barPlots.py:76]: Is this comment relevant to this line?

File: python/lsst/analysis/tools/actions/plot/barPlots.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    def makePlot(
+        self, data: KeyedData, plotInfo: Mapping[str, str] = None, **kwargs  # type: ignore
+    ) -> Figure:
+        """Make an N-panel plot with a user-configurable number of bar graphs
+        displayed in each panel.
+
+
+        Parameters
+        ----------
+        data : `pandas.core.frame.DataFrame`
```
Comment (sr525) [inline python/lsst/analysis/tools/actions/plot/barPlots.py:87]: data doesn't need to be a dataframe it should be able to take any form of keyed data.

File: python/lsst/analysis/tools/actions/plot/barPlots.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+                assigned_labels.append(single_label)
+                assigned_colors.append(col[i])  # Assign color from color cmap
+
+            i += 1
+
+        return x_values, assigned_labels, assigned_colors
+
+    def _sortBarBins(self, x_values, assigned_labels, assigned_colors):
+        """Sorts the existing x_values, assigned_labels,
+        and assigned_colors.Sorts x_value from lowest to
```
Comment (sr525) [inline python/lsst/analysis/tools/actions/plot/barPlots.py:285]: typo, colors.Sorts

**Comment by sr525** [comment]
Can an see an example of what this outputs?

File: python/lsst/analysis/tools/actions/plot/barPlots.py (hunk 1 above)
Comment (bsmartradio) [reply python/lsst/analysis/tools/actions/plot/barPlots.py:76]: No, a holdover from a different piece of code. Removed now.

**Comment by bsmartradio** [comment]
Here is a test I ran in a notebook. The first image is just two different vectors, an unassociated sources and associated sources, which is what I intend to use it for. The second graph is a test with multiple different vectors, though it's all drawing from the same data, just repeated to test the multiple bars portion.
Screen Shot 2022-11-08 at 10 44 17 AM

**Comment by sr525** [comment]
How does this look when it is run from the command line?

**Comment by sr525** [comment]
Does it have enough space for the plot info at the top?

**Comment by bsmartradio** [comment]
> 

Yes, it looks like this with the info at the top 
Screen Shot 2022-11-15 at 9 46 59 AM

**Comment by bsmartradio** [comment]
Here is what it looks like using ap_verify_ci_hits2015 on branch DM-36263. 

```
pipetask run  -b demo_pipeline/repo/butler.yaml -i ap_verify-output --output test_metric_plot --pipeline /sdf/home/s/smart/u/code/analysis_tools/pipelines/testPipe.yaml -d "detector=5 AND instrument='DECam'" --register-dataset-types
```
sourcesOutput_g_analysisPlots_DECam_g_g_DECam_SDSS_c0001_4720_0_1520_0_419802_test_metric_plot_20221117T153256Z

File: python/lsst/analysis/tools/actions/plot/barPlots.py [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+        Returns
+        -------
+        fig : `matplotlib.figure.Figure`
+            The resulting figure.
+
+        """
+
+        # set up figure
+        fig = plt.figure(dpi=800)
```
Comment (sr525) [inline python/lsst/analysis/tools/actions/plot/barPlots.py:116]: Do we need this high a dpi? Do the figures come out very large?

Comment (bsmartradio) [reply python/lsst/analysis/tools/actions/plot/barPlots.py:116]: It probably doesn't. I increased the dpi to test something Meredith was asking about and never changed it back. I'll lower it.
